sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationAsyncClientTest.java

```diff
@@ -91,17 +97,21 @@ public void addConfigurationSetting() {
     /**
      * Tests that we cannot add a configuration setting when the key is an empty string.
      */
-    @Test
-    public void addConfigurationSettingEmptyKey() {
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void addConfigurationSettingEmptyKey(HttpClient httpClient, ConfigurationServiceVersion serviceVersion) {
```

Comment:
Not a huge fan that every test will need to become parameterized to test each client, are there any ways to abstract this into Azure Core Test or does JUnit have the ability to parameterize an entire test run?

Reply:
JUnit 4 did support this, but as far as I know, JUnit 5 does not - you now parameterise each test separately.

Reply:
> First, I'd like to see a description so I can understand why these changes are necessary. Second, I don't believe that pom.xml isn't the correct place to add a core dependency.

Thanks for suggestion! Added the description.